rest-proxy-core/src/main/groovy/edu/wisc/my/restproxy/model/ProxyReport.groovy

```diff
@@ -0,0 +1,26 @@
+package edu.wisc.my.restproxy.model
+
+import groovy.transform.Canonical
+import groovy.transform.CompileStatic
+import groovy.transform.builder.Builder
+import groovy.transform.builder.SimpleStrategy
+
+/**
+ * ProxyReport : Utilized to report back when was the last success/failure of each key
+ */
+
+@Builder(builderStrategy = SimpleStrategy)
+@Canonical
+@CompileStatic
+public class ProxyReport {
+  public ProxyReport(String key){
+    this.key = key;
+  }
+  String key;
+  Map<String, Long> lastResultMap = new HashMap<>();
```

Comment:
I'd prefer a comment here with excruciating clarity about what the keys and values of this map represent.